### ISSUE TITLE
Poetry cannot install msal-extensions

### ISSUE THREAD

**Issue by chlowell** [AzureAD/microsoft-authentication-extensions-for-python#62]
The [multiple portalocker requirements](https://github.com/AzureAD/microsoft-authentication-extensions-for-python/blob/35641c37de1f0bbb214627769e5847e74808ad55/setup.py#L28)
```
portalocker~=1.6;platform_system=='Windows'",
portalocker~=1.0;platform_system!='Windows'"
````
make `msal-extensions` impossible to install with Poetry due to python-poetry/poetry#1238. Although this is a known Poetry bug, it could presumably be avoided if `msal-extensions` didn't have these platform-specific requirements. Given that the latest `portalocker` is 1.7.1, are they still necessary?

**Comment by rayluo** [comment]
@bluca You and us tried very hard (at [exact 3 months ago](https://github.com/AzureAD/microsoft-authentication-extensions-for-python/pull/49)) to allow using prior 1.6.0 `portalocker` on older Linux distributions. What if we bump it up to `portalocker~=1.6` on all platforms this time?

**Comment by bluca** [comment]
I think all versions can be supported here, thus relaxing the requirements in setup.py. For example it looks like versions 1.7.0 and 1.7.1 just added minor fixes, no compatibility was removed. So you could have >= 1.0 on Linux and >= 1.6 on Windows (I think there were Windows-specific fixes in that release)

**Comment by rayluo** [comment]
* Discussed with @bluca off-line. It would be fine if we bump our dependency to consistent `portalocker~=1.6` on all platforms. Luca explained that a Linux package maintainer could choose to ignore `setup.py` and still use an older version of dependency. (For example, as of today, stable [portalocker on Debian](https://packages.debian.org/search?keywords=python3-portalocker&searchon=names&suite=all&section=all) is still at 1.3, similar [situation on Ubuntu](https://packages.ubuntu.com/search?keywords=python3-portalocker&searchon=names&suite=all&section=all)). But such an outside-of-`setup.py` dependency may not continue to work in the long run.

* Alternatively, [this issue can probably be resolved by using `poetry 1.1.0a2`, currently in preview, and stable version is coming at Aug](https://github.com/python-poetry/poetry/issues/1238#issuecomment-662675156).

We incline to not make any changes from our side.

**Comment by alexkyllo** [comment]
I'm having this issue again with poetry version 1.1.2

When I try to do `poetry add azure-identity` I get this infinite loop:
```

1: selecting azure-identity (1.4.1)
   1: fact: azure-identity (1.4.1) depends on azure-core (>=1.0.0,<2.0.0)
   1: fact: azure-identity (1.4.1) depends on cryptography (>=2.1.4)
   1: fact: azure-identity (1.4.1) depends on msal (>=1.3.0,<2.0.0)
   1: fact: azure-identity (1.4.1) depends on msal-extensions (>=0.2.2,<0.3.0)
   1: fact: azure-identity (1.4.1) depends on six (>=1.6)
   1: selecting azure-identity (1.4.1)
```

**Comment by rayluo** [comment]
@alexkyllo Previous [conversation in that Poetry issue](https://github.com/python-poetry/poetry/issues/1238#issuecomment-662675156) got confirmed that Poetry fixed the problem since 1.1.0a2. If you find this issue again in their later version, it sounds like a regression. Would you consider create/reopen an issue there?
